subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/DefaultFileSystemLocationFingerprint.java

```diff
@@ -45,8 +48,82 @@ private static HashCode hashForType(FileType fileType, HashCode hash) {
         }
     }
 
+    @Override
+    public final void appendToHasher(Hasher hasher) {
+        hasher.putString(getNormalizedPath());
+        hasher.putHash(getNormalizedContentHash());
+    }
+
+    @Override
+    public FileType getType() {
+        if (normalizedContentHash == DIR_SIGNATURE) {
```

Comment:
I wouldn't have removed type. What benefit do we get from removing it?

Reply:
Smaller memory footprint? It's just redundant information anyway. Why would you keep it? (Note that we still have `getType()`.)

Reply:
IMO the class is easier to understand when we don't throw away the type in the constructor just to recalculate the information here again.

Reply:
I disagree. For me having the type represented by both `type` and the content hash was more confusing. I think the semantics of `getType()` is pretty clear, and that's what we use to determine the type everywhere. The "calculation" is also pretty simple.

OTOH, we store potentially many thousands or even hundreds of thousands of these objects during a build; not having to add a field with redundant information can have a significant impact on memory usage.

Reply:
👍